pandas-stubs/core/frame.pyi

```diff
@@ -250,7 +250,7 @@ class DataFrame(NDFrame, OpsMixin):
     def iterrows(self) -> Iterable[tuple[Hashable, Series]]: ...
     def itertuples(
         self, index: _bool = ..., name: _str | None = ...
-    ) -> Iterable[tuple[Any, ...]]: ...
+    ) -> Iterable[Any]: ...
```

Comment:
I'm wondering if we could do something like this:
```python
class _PandasNamedTuple(tuple[Any,...]):
    def __getattr__(self, field: str) -> Any: ...
```
Then have `itertuples()` return `Iterable[_PandasNamedTuple]`

Then at least you know you're getting a tuple.

Reply:
Nice solution!